src/overcooked_ai_py/agents/benchmarking.py

```diff
@@ -20,83 +19,65 @@ class AgentEvaluator(object):
         env_params (dict): params for creation of an OvercookedEnv
-        mdp_fn_params (dict): params to setup random MDP generation
-        force_compute (bool): whether should re-compute MediumLevelPlanner although matching file is found
-        mlp_params (dict): params for MediumLevelPlanner
+        mdp_fn (callable function): a function that can be used to create mdp
+        force_compute (bool): whether should re-compute MediumLevelActionManager although matching file is found
+        mlam_params (dict): the parameters for mlam, the MediumLevelActionManager
+        debug (bool): whether to display debugging information on init
         """
         assert type(mdp_params) is dict, "mdp_params must be a dictionary"
-
-        if mdp_fn_params is None:
+        assert (mdp_params != {} and env_params != {}) or mdp_fn != None, "either evaluate from params or fn"
+        env_params["mlam_params"] = mlam_params
+        if mdp_fn is None:
             mdp = OvercookedGridworld.from_layout_name(**mdp_params)
             self.mdp_fn = lambda: mdp
             self.env = OvercookedEnv.from_mdp(mdp, **env_params)
         else:
-            self.mdp_fn = LayoutGenerator.mdp_gen_fn_from_dict(mdp_params, **mdp_fn_params)
-            self.env = OvercookedEnv(self.mdp_fn, **env_params)
-        
-        self.force_compute = force_compute
-        self.debug = debug
-        self.mlp_params = mlp_params
-        self._mlp = None
-
-    @property
-    def mlp(self):
-        assert not self.env.variable_mdp, "Variable mdp is not currently supported for planning"
-        if self._mlp is None: 
-            if self.debug: print("Computing Planner")
-            self._mlp = MediumLevelPlanner.from_pickle_or_compute(self.env.mdp, self.mlp_params, force_compute=self.force_compute)
-        return self._mlp
-
-    def get_params(self):
-        return {
-            "mdp_params": self.mdp.mdp_params,
-            "env_params": self.env.env_params,
-            "mdp_fn_params": self.mdp_fn_params,
-            "force_compute": self.force_compute,
-            "mlp_params": self.mlp_params,
-            "debug": self.debug
-        }
-
-    def evaluate_random_pair(self, num_games=1, all_actions=True, display=False):
-        agent_pair = AgentPair(RandomAgent(all_actions=all_actions), RandomAgent(all_actions=all_actions))
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
+            # infinite mdp
+            if 'num_mdp' not in env_params or env_params['num_mdp'] == -1:
+                self.mdp_fn = mdp_fn
+                self.env = OvercookedEnv(self.mdp_fn, **env_params)
+            else:
+                num_mdp = env_params['num_mdp']
+                assert num_mdp > 0, "invalid number of mdp"
+                self.mdp_lst = [mdp_fn() for _ in range(num_mdp)]
+                self.mdp_fn = lambda : np.random.choice(self.mdp_lst)
+                self.env = OvercookedEnv(self.mdp_fn, **env_params)
```

Comment:
Regarding the if/else logic in the init, we might want to have this handled with `@classmethod`s here too. Given that OvercookedEnv now works based off of `mdp_generating_fn`, we might even be able to deprecate the `mdp_fn` `__init__` signature parameter, and clean things up here more.

For example, one downside of this approach is that having `self.mdp_lst = [mdp_fn() for _ in range(num_mdp)]` is making strong assumptions on how the mdp_fn() will return different mdps sequentially at each call, and the following line is forcing a specific sampling function (which i guess in the case of the AgentEvaluator would probably usually make sense, but it would be nice to have this logic outside of the class or in a classmethod rather than in conditional logic in an init so that it's flexible to customize for strage use-cases).

Reply:
After exploring many option, I think I have one method figured out, and I am adding tests to make sure it is robust. Will update in the next commit